pkgs/by-name/ze/zerotierone-desktop-ui/package.nix

```
@@ -0,0 +1,95 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, meson
+, ninja
+, pkg-config
+, gnumake
+, gcc
+, zlib
+, freetype
+, fontconfig
+, gtk3
+, libappindicator
+}:
+
+
+let
+  pname = "zerotierone-desktop-ui";
+  version = "2024.9.0"; # Infrequent updates and does not seem to be released often
+
+  src = fetchFromGitHub {
+    owner = "zerotier";
+    repo = "DesktopUI";
+    rev = "452eca3270f84ef4ac81c955f4777c8f1a82b988";
+    hash = "sha256-1gYF9v7sxCO7+pBRBSCYSRaoHEW3Xvx7nh28TsERZy8=";
+  };
+in
+rustPlatform.buildRustPackage rec {
+  inherit pname version src;
+
+  patches = [ ./gcc14-compile-error-fix.patch ];
+
+  cargoLock = {
+    lockFile = "${src}/Cargo.lock";
+    outputHashes = {
+      "mac-notification-sys-0.5.0" = "sha256-VIFkZWksV7CxeVAiXWtTdJe1v4keoiQSSnZVv8uZTeo=";
+    };
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    gnumake
+  ];
+
+  buildInputs = [
+    gcc
+    zlib
+    freetype
+    fontconfig
+    gtk3
+    libappindicator
+  ];
+
+  configurePhase = ''
+    echo "=== Building libui-ng ==="
+    cd libui-ng
+    meson setup build --buildtype=release --default-library=static --backend=ninja
+    ninja -C build
+    cd ..
+
+    echo "=== Building tray ==="
+    cd tray
+    make clean
+    make zt_lib
+    cd ..
+  '';
+
+  buildPhase = ''
+    cargo build --release
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share/applications
+    mkdir -p $out/share/zerotier
+    cp target/release/zerotier_desktop_ui $out/bin/
+    cp ZeroTierIcon.png $out/share/zerotier/
+    cp zerotier-ui.desktop $out/share/applications/
+    substituteInPlace $out/share/applications/zerotier-ui.desktop \
+      --replace "/usr/bin/zerotier_desktop_ui" "$out/bin/zerotier_desktop_ui" \
+      --replace "/usr/share/zerotier/ZeroTierIcon.png" "$out/share/zerotier/ZeroTierIcon.png"
+  '';
+
+  meta = with lib; {
```

```suggestion
  meta = {
```
Avoid nesting `with` expressions.